[#251] Reorganize readme (#253)
serokell/tezos-packaging: README.md

## Table of contents

> * [Getting binaries](#getting-binaries).

Reviewer: If you take my previous suggestion this could have sub-entries IMO.

* [Systemd services with Tezos binaries](#systemd).

> * [Launchd services with Tezos binaries](#systemd).

Reviewer: I'm thinking that perhaps the `Launchd` document should be moved inside the `macOS` one.

From its content we only provide the former in the formulae for the latter, plus it makes sense to me that only `macOS` users might be interested in it.

For `sysyemd` packages given that they are provided for both native releases and can be build from scratch I'd leave those instructions in a separate doc (as they are now), but maybe instead of linking it from here we could add a link in the two distro docs as well as under the "other distros" section below?

> * [Setting up baking instance on Ubuntu](#baking-on-ubuntu).

Reviewer: Maybe change it to:
```suggestion
* [Setting up a node and/or baking on Ubuntu](#baking-on-ubuntu).
```

Reviewer: You updated the link but not the header's content.

* [Building instructions](#building).

> 

Reviewer: Maybe contributing and about should have an entry here too?

<a name="getting-binaries"></a>
## Getting Tezos binaries

The following distributions are supported by the `tezos-packaging`:

> * Ubuntu. Install packages using `apt-get` following [these instructions](./docs/distros/ubuntu.md).

Reviewer 2: ```suggestion
* **Ubuntu**. Install packages using `apt-get` following [these instructions](./docs/distros/ubuntu.md).
```

>   This option also supports Raspberry Pi OS, see [this part of the doc](./docs/distros/ubuntu.md#raspberry).
> * Fedora. Install packages using `dnf` or `yum` following [these instructions](./docs/distros/fedora.md).
> * Other linux distros can use static binaries from the [latest release](https://github.com/serokell/tezos-packaging/releases/latest).
> * macOS. Install formulaes using `brew` following [these instructions](./docs/distros/macos.md).

Reviewer 2: ```suggestion
  This option also supports **Raspberry Pi OS**, see [this part of the doc](./docs/distros/ubuntu.md#raspberry).
* **Fedora**. Install packages using `dnf` or `yum` following [these instructions](./docs/distros/fedora.md).
* Other linux distros can use static binaries from the [latest release](https://github.com/serokell/tezos-packaging/releases/latest).
* **macOS**. Install formulaes using `brew` following [these instructions](./docs/distros/macos.md).
```
I suggest to add bold emphasis, otherwise I found it hard to skim through.

> * **macOS**. Install formulaes using `brew` following [these instructions](./docs/distros/macos.md).

Reviewer: A mostly aesthetic suggestion I'd make is to make a heading for each entry, which is itself the link, and only put a small description below it.

Here my suggestion for this whole section would be:
```
<a name="getting-binaries"></a>
## Getting Tezos binaries

The following distributions are supported by the `tezos-packaging`:

### [macOS](./docs/distros/macos.md)

`brew` formulae and taps for stable releases and release candidates.

### [Ubuntu](./docs/distros/ubuntu.md)

Native Ubuntu packages that can be installed with `apt-get` from a PPA.

### [Rasperry Pi OS](./docs/distros/ubuntu.md#raspberry)

On some Raspberry systems Ubuntu packages can be installed using `apt-get`.

### [Fedora](./docs/distros/fedora.md)

Native Fedora packages that can be installed using using `dnf` or `yum`.

### [Static binaries and other linux distros](https://github.com/serokell/tezos-packaging/releases)

Prebuilt static binaries can be downloaded directly from the [latest release](https://github.com/serokell/tezos-packaging/releases/latest).
```

I've see this done in some other READMEs and I think it can be easier to read.

## Table of contents

* [Getting binaries](#getting-binaries).
  * [Ubuntu](#ubuntu)
  * [Raspberry Pi OS](#raspberyy)
  * [Fedora](#fedora)
  * [Other Linux distros](#linux)
  * [macOS](#macOS)
* [Setting up a node and/or baking on Ubuntu](#baking-on-ubuntu).
* [Building instructions](#building).
* [Contribution](#contribution).
* [About Serokell](#about)

<a name="getting-binaries"></a>
## Getting Tezos binaries

The following distributions are supported by the `tezos-packaging`:

<a name="ubuntu"></a>
### [Ubuntu](./docs/distros/ubuntu.md)

Native Ubuntu packages that can be installed with `apt-get` from a PPA.

<a name="raspberry"></a>
### [Raspberry Pi OS](./docs/distros/ubuntu.md#raspberry)

Some Raspverry systems also support packages that can be installed using `apt-get`.

<a name="fedora"></a>
### [Fedora](./docs/distros/fedora.md)

Native Fedora packages that can be installed using using `dnf` or `yum`.

<a name="linux"></a>

> ### [Static binaries and other linux distros]

Reviewer: Uh, this without a link _and_ between square brackets is odd, why not link this to the `/releases` page?

Author: uh, I somehow lost the link :confused:
